terraform/modules/dandiset_bucket/main.tf

```diff
@@ -110,3 +110,47 @@ resource "aws_iam_user_policy" "dandiset_bucket_owner" {
     ]
   })
 }
+
+resource "aws_s3_bucket_policy" "dandiset_bucket_policy" {
+  provider = aws
+
+  bucket = aws_s3_bucket.dandiset_bucket.id
+  policy = jsonencode({
+    Version = "2008-10-17"
+    Statement = [
+      {
+        Principal = {
+          AWS = var.heroku_user.arn
+        }
+        Action = [
+          "s3:Get*",
+          "s3:List*",
+          "s3:Delete*",
```

Comment:
I'm not totally sure of the reason for the split between these two statements (or whatever the proper term is), but it seems weird that `Delete` is lumped in here with `Get` and `List`. Why is that?

Reply:
One statement gives `heroku_user` permission to do anything as long as they provide the `X-AMZ-ACL=bucket-owner-full-control` header. This is required for uploads and updates to be owned by the bucket owner instead of the uploader. However, many endpoints don't require that header (specifically Gets, Lists, and Deletes), and `heroku_user` also needs permission to perform those operations.

It's possible there's a better way to set this up, but I don't know enough about ACLs to know what it is.